server/pbench/bin/index-pbench

```
@@ -322,6 +330,168 @@ class ResultData(object):
+
+                ds = (results['latency']['usec'][i],
+                      results['throughput']['trans_sec'][i])
+                data.append(ds)
+            # print("data (len, type, value) = ", len(data), type(data))
+            # pp.pprint(data)
+        elif 'throughput' in results:
+            throughput = True
+            for i in range(len(results['throughput']['Gb_sec'])):
+                ts = []
+                ds = []
+                ts.extend(zip(results['throughput']['Gb_sec'][i]['timeseries'].copy()))
+                tseries.append(ts)
+                convert_to_float(tseries)
+                del results['throughput']['Gb_sec'][i]['timeseries']
+                ds = (results['throughput']['Gb_sec'][i],)
+                data.append(ds)
+            # print("data (len, type) = ", len(data), type(data))
+            # pp.pprint(data)
+        else:
+            # impossible?
+            return
+
+        # We now repackage everything in ES-indexable form:
+        # Each document will have metadata, a single timestamp, and the data associated with that
+        # timestamp, plus the specific metadata for that data (e.g the client/server pair)
+
+        # pp.pprint(md)
+        # pp.pprint(results['latency'])
+        # pp.pprint(results['throughput'])
+        # for ts in tseries:
+        #     convert_to_float(ts)
+        #     for i in range(10):
+        #         print(ts[i])
+        #     print()
+        # return
+
+        # print(len(tseries))
+        #import pdb; pdb.set_trace()
+
+        for i in range(len(tseries)):
+            if latency:
+                dat = {
+                    "latency_desc": data[i][0],
+                    "throughput_desc": data[i][1]
+                }
+            elif throughput:
+                dat = {"throughput_desc": data[i][0]}
+            else:
+                # impossible?
+                return
```

Can we raise an exception instead?